pkg/client/add_events.go

```diff
@@ -95,6 +89,35 @@ func (client *DataSetClient) AddEvents(bundles []*add_events.EventBundle) error
 	return nil
 }
 
+// IsProcessingData returns True if there are still some unprocessed data.
+// False otherwise.
+func (client *DataSetClient) IsProcessingData() bool {
+	return client.buffersEnqueued.Load() > client.buffersProcessed.Load()
+}
+
+// Finish stops processing of new events and waits until all the data that are
+// being processed are really processed.
+func (client *DataSetClient) Finish() {
```

Comment:
Is there a way for the end user to utilize defer to ensure this function is always called at the end or similar?